### DISCUSSION TITLE
UPPAAL 4.1.26 server exited with error 3 when requesting license

### DISCUSSION THREAD

**Discussion by linhngocle2000** [UPPAALModelChecker/UPPAAL-Meta#274] [Q&A] [answered]
I am using uppaal64-4.1.26-2. When I run uppaal.exe, I get the error
![pic1](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/62800032/829cde7d-9301-41b1-8d9d-572929fcb613)

When I try to run server.exe directly in the terminal, I get the error
![pic2](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/62800032/0c74418b-b004-4427-a62b-ab888e036e94)

Due to the error, I cannot check model properties. Would someone please help me? Thank you very much.

**Comment by mikucionisaau** [comment]
The issue is at `it.uu.se` server configuration, I am trying to fix this.

**Reply by hendrikgoettmann** [reply]
Hi Marius,

is there any update regarding this issue?
For certain reasons, our department uses Uppaal Stratego 4.1.x instead of Uppaal 5.
For instance, we cannot switch to Uppaal 5 due to issue [https://github.com/UPPAALModelChecker/UPPAAL-Meta/issues/273](https://github.com/UPPAALModelChecker/UPPAAL-Meta/issues/273).
Thus, we would be very happy if we can get a small hint on the current state of this issue.

Thanks in advance.

**Reply by mikucionisaau** [reply]
see update in the next answer

**Comment by mikucionisaau** [comment, marked as answer]
Hi,
I have created a utility `uppaal-lease` which creates `license.txt` file for UPPAAL-4 and other old UPPAAL releases.
The license needs to be recreated every 12-14 days.

1) Download `uppaal-lease` utility using one of the links below.
2) Extract the archive and double-click to run `uppaal-lease.exe`, which will create `license.txt` file in the current directory as well as in current user's configuration where UPPAAL can pick it up: 
   - Windows: `%LOCALAPPDATA%/uppaal/license.txt`
   - MacOS and Linux: `$HOME/.uppaal/license.txt`
3) For UPPAAL TIGA, please copy the `license.txt` file into TIGA installation directory.

Windows:
- [uppaal-lease for Windows x86_64](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-win-x86_64.zip)

Linux:
- [uppaal-lease for Fedora 40 x86_64 (GLIBC-2.39)](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-fedora-40.zip)
- [uppaal-lease for Ubuntu 24.04 x86_64 (GLIBC-2.39)](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-ubuntu-24.04.zip)
- [uppaal-lease for Ubuntu 22.04 x86_64 (GLIBC-2.35)](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-ubuntu-22.04.zip)
- [uppaal-lease for Ubuntu 20.04 x86_64 (GLIBC-2.31)](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-ubuntu-20.04.zip)
- [uppaal-lease for Linux aarch64](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-linux-aarch64.zip)

MacOS:
- [uppaal-lease for MacOS x86_64](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-darwin-x86_64.zip)
- [uppaal-lease for MacOS aarch64](https://download.uppaal.org/uppaal-leases-2.6/uppaal-leases-2.6.0/uppaal-leases-2.6.0-darwin-aarch64.zip)

**Comment by hendrikgoettmann** [comment]
Hi, 

I can confirm that it works on my machine with Windows 11 and Ubuntu 20.04.
Thank you very much!
